tgui/packages/tgui/interfaces/SelectEquipment.js

```diff
@@ -0,0 +1,149 @@
+import { useBackend, useLocalState } from '../backend';
+import { createSearch } from 'common/string';
+import { Box, Button, Tabs, Section, Input, Stack, Flex, Divider } from '../components';
+import { Window } from '../layouts';
+
+export const SelectEquipment = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    name,
+    outfits,
+    icon64,
+  } = data;
+
+
+  // search bar
+  const [
+    searchText,
+    setSearchText,
+  ] = useLocalState(context, 'searchText', '');
+  const searchFilter = createSearch(searchText, entry =>
+    (entry[0] + entry[1])
+  );
+  const searchBar
+    = (<Input
+      fluid
+      autoFocus
+      placeholder="Search"
+      value={searchText}
+      onInput={(e, value) => setSearchText(value)}
+      mb={1} />);
+
+
+  // outfit tabs; mapped and named from the data sent by ui_static_data
+  const [
+    tabIndex,
+    setTabIndex,
+  ] = useLocalState(context, 'tab-index', "General");
+
+  const OutfitTab = props => {
+    const { name, ...rest } = props;
+    return (
+      <Tabs.Tab
+        selected={tabIndex === name}
+        onClick={() => setTabIndex(name)}
+        {...rest}>
+        {name}
+      </Tabs.Tab>
+    );
+  };
+
+  const outfitCategories = Object.keys(outfits);
```

Comment:
You can send data better so that you don't need to do this but honestly not the end of the world